### DISCUSSION TITLE
Is building a separate REST API more secure than using API Routes?

### DISCUSSION THREAD

**Discussion by THPubs** [vercel/next.js#63466] [Help]
### Summary

Hi, I'm building an enterprise application which handles sensitive data. For this I have two routes... Use NextJS API routes and deploy the backend and the front end in the same server or build a separate backend server. Some argue that hosting both the backend and the front end in the same server is a security risk because hackers have only one server to hack in order to get direct access to the database. Have anyone built this kind of apps in full stack NextJS?

**Comment by Vijaykv5** [comment]
Hi @THPubs 

I personally use API Routes at the moment for my side projects, but API routes in nextjs dont seem very mature for me.

Yes! as you said, separating the frontend and backend can add a layer of security, reducing the risk of direct database access through frontend vulnerabilities.

This [blog](https://nextjs.org/blog/security-nextjs-server-components-actions) may help!
